src/mds/MDCache.cc

```diff
@@ -312,6 +312,7 @@ void MDCache::remove_inode(CInode *o)
     inode_map.erase(o->ino());
   } else {
     o->item_caps.remove_myself();
+    o->item_to_flush.remove_myself();
```

Comment:
@dparmar18 It seems the inode in both `inode_map` and `snap_inode_map` could call `mark_need_snapflush_inode()`.

Reply:
`Locker::mark_need_snapflush_inode` has this assertion `ceph_assert(in->last != CEPH_NOSNAP);` while the condition for removal of inode from `inode_map` is completely opp 
```
if (o->last == CEPH_NOSNAP) {
    inode_map.erase(o->ino());
  }
```

so i think they `inode_map` and `snap_inode_map` aren't relevant with each other in this case, or am i getting this completely wrong?

Reply:
Yeah, I didn't dig it further yet. And just check the `cow_inode()` and it seems doing the same check in both inode maps. Maybe I was wrong.